src/libraries/System.Data.Common/src/System/Data/XMLSchema.cs

```diff
@@ -26,6 +26,7 @@ internal static TypeConverter GetConverter(Type type)
             return TypeDescriptor.GetConverter(type);
         }
 
+        [RequiresUnreferencedCode(DataSet.RequiresUnreferencedCodeMessage)]
```

Comment:
Perhaps a more accurate message here would be that the type of instance and individual attributes cannot be statically analyzed so their members may get trimmed, which in this case would break as it seems like we try to get all properties from instance type.